apps/rush-lib/src/cli/actions/AddAction.ts

```diff
@@ -116,16 +120,25 @@ export class AddAction extends BaseRushAction {
       return Promise.reject(new Error(`The SemVer specifier "${version}" is not valid.`));
     }
 
-    return new PackageJsonUpdater(this.rushConfiguration, this.rushGlobalFolder).doRushAdd({
-      currentProject: currentProject,
-      packageName: packageName,
-      initialVersion: version,
-      devDependency: this._devDependencyFlag.value,
-      updateOtherPackages: this._makeConsistentFlag.value,
-      skipUpdate: this._skipUpdateFlag.value,
-      debugInstall: this.parser.isDebug,
-      rangeStyle: this._caretFlag.value ? SemVerStyle.Caret
-        : (this._exactFlag.value ? SemVerStyle.Exact : SemVerStyle.Tilde)
-    });
+    let projects: RushConfigurationProject[];
+    if (this._allFlag) {
+      projects = Object.values(this._rushConfiguration.projectsByName);
+    } else {
+      projects = [currentProject];
+    }
+
+    for (const project of projects) {
+      return new PackageJsonUpdater(this.rushConfiguration, this.rushGlobalFolder).doRushAdd({
```

Comment:
What happens if a project already has the dependency being added?

Reply:
I will take a look!

Reply:
How does `rush add`currently manages this scenario?

Reply:
I don't see it in the source code of the action nor the logic. I must be missing something 🤔

Reply:
The current underling logic of rush add does a `noop` when the package already exists. I believe it's safe to rely on this. We are not affecting this core logic, just running the same logic on multiple packages. Let me know if I'm wrong! I'm just learning about this.

Reply:
Cool - sounds good to me.